extensions/2.0/Vendor/EXT_feature_metadata/README.md

```diff
@@ -0,0 +1,1691 @@
+          "extensions": {
+            "EXT_feature_metadata": {
+              "featureIdAttributes": [
+                {
+                  "featureTable": "trees",
+                  "featureIds": {
+                    "attribute": "_FEATURE_ID_0"
+                  }
+                }
+              ]
+            }
+          }
+        }
+      }
+    }
+  ]
+}
+```
+
+## Feature Metadata
+
+Feature metadata is structured according to a **schema**. A schema has a set of **classes** and **enums**. A class contains a set of **properties**, which may be numeric, boolean, string, enum, or array types.
+
+A **feature** is a specific instantiation of class containing **property values**. Property values are stored in either a **feature table** or a **feature texture** depending on the use case. Both formats are designed for storing property values for a large number of features.
+
+**Statistics** provide aggregate information about the metadata. For example, statistics may include the min/max values of a numeric property for mapping property values to color ramps or the number of enum occurrences for creating histograms.
+
+By default properties do not have any inherent meaning. A property may be assigned a **semantic**, an identifier that describes how the property should be interpreted. The full list of built-in semantics can be found in the [Cesium Metadata Semantic Reference](https://github.com/CesiumGS/3d-tiles/tree/3d-tiles-next/specification/Metadata/Semantics). Model authors may define their own application- or domain-specific semantics separately.
+
+This extension implements the [Cesium 3D Metadata Specification](https://github.com/CesiumGS/3d-tiles/tree/3d-tiles-next/specification/Metadata), which describes the metadata format in full detail.
+
+### Schemas
+
+A schema defines a set of classes and enums used in a model. Classes serve as templates for features - they provide a list of properties and the type information for those properties. Enums define the allowable values for enum properties. Schemas are defined in full detail in the [Schemas](https://github.com/CesiumGS/3d-tiles/tree/3d-tiles-next/specification/Metadata#schemas) section of the [Cesium 3D Metadata Specification](https://github.com/CesiumGS/3d-tiles/tree/3d-tiles-next/specification/Metadata).
+
+A schema may be embedded in the extension directly or referenced externally with the `schemaUri` property. Multiple glTF models may refer to the same external schema to avoid duplication.
+
+Schemas may be given a `name`, `description`, and `version`.
+
+### Feature Tables
+
+A feature table stores property values in a parallel array format. Each property array corresponds to a class property. The values contained within a property array must match the data type of the class property. Furthermore, the set of property arrays must match one-to-one with the class properties. There is one exception - if a property is optional the feature table may omit that property.
```

Comment:
hm that's a good point, if a class has both a feature table and feature texture, you wouldn't need a column for the feature texture properties. Either that or restrict things so you can't use the same class for both a table and a texture. 

I guess the one thing is do we have any good use cases for using both types of encodings with a single class?

Reply:
> I guess the one thing is do we have any good use cases for using both types of encodings with a single class?

I don't have any use cases in mind, and the two encodings don't really share the same concept of a "feature", maybe it's better to restrict this...

Reply:
@lilleyse or @sanjeetsuhag any thoughts/use cases on this? I'm leaning towards just restricting it so a class can either be for a texture XOR a feature table.

Reply:
Hm I hadn't though much about how optional properties might be handled with feature textures. Though I'm not sure this is a strong enough reason to disallow feature tables and feature textures from using the same class. It seems natural that two different encoding might happen to use the same class.

Reply:
Maybe we need to define what a "feature" really is. To me the concept of feature needs to be defined at the geometry level rather than the encoding level because a feature is something that has some sort of physical manifestation.

Another way of thinking of it is that a feature table doesn't actually contain features, it contains feature properties. The actual feature is the group of vertices on the mesh. This could also be extended to feature textures. The surface of the mesh covered by a single texel is the feature and the texel itself contains the feature properties. However it's a little abstract to think of feature textures that way.

It could be that "feature table" and "feature texture" are the wrong names. Instead "metadata table" and "metadata texture"? Or "property table" and "property texture"?

Reply:
> It seems natural that two different encoding might happen to use the same class.

That's OK – the idea that a class schema could be implemented by a feature table in some cases, or baked to textures and implemented by a feature texture in another case, seems fine, and it's appropriate to reuse the class.

I think the main problem is that the meaning of an "optional property" is currently defined only under the Feature Tables section, so there's ambiguity about what happens with textures involved. Can a single primitive implement a class schema with 6 required properties by using a feature table with 3/6 properties and a feature texture set with 3/6 properties? That usage seems problematic because (1) the 'feature' granularity is not consistent, and (2) neither the table nor the texture set satisfy the most intuitive interpretation of optional vs. required properties. 

***

On its own the term "feature table" makes sense to me. The term "feature texture" I'm having a bit more trouble with, because the meaning of a feature is not fully defined in that context, and because it isn't necessarily a single texture. The names "property table" and "property texture" also sound good to me, aside from that same inconsistency with the singular "texture". 

Two ideas:

(a) Adjust the name and/or description to set expectations for multiple textures, e.g. "feature texture set" or "group."

(b) Adjust the schema so each "feature texture" is a single texture, and of course a primitive may have several of them. E.g.:

```
"featureTextures": [{
	"index": 0,
	"texCoord": 0,
	"properties": {
		"heatLoss": "r",
		"heatGain": "g"	
	}
}]
```

Reply:
Proposed changes based on (b) in https://github.com/CesiumGS/glTF/pull/14.